### DISCUSSION TITLE
Projects: Roles: Prevent Write Role To Add or Update Views (= add maintainer role)

### DISCUSSION THREAD

**Discussion by BenjaminYde** [community/community#52623] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

Currently the write role can change the views and save them, but only a maintainer should be able to do this!
As you can see on the picture below the setting is write, but this means that every member can also add views (table,board,...) / change them, ...

![image](https://user-images.githubusercontent.com/44135938/231676914-5c703ab9-8865-4718-887f-5e2408cb9678.png)

What I want is the following:

- Members can be added to the project
- Members can add issues inside the project (and update them)
- Members cannot add / edit views (only maintainers should be able to)

**Comment by opegamanalitarok** [comment]
To restrict the ability to change and save views in a GitHub repository to only maintainers, you can adjust the repository's settings to modify the permissions of the write role. Here's how to do it:

1. Go to the repository's settings by clicking on the "Settings" tab on the repository's main page.
2. Click on "Manage access" on the left sidebar.
3. Find the "Write" role in the list of collaborators and click on the pencil icon next to it to edit the role's permissions.
4. Under the "Repository permissions" section, uncheck the box next to "Manage repository".
5. Under the "Actions permissions" section, uncheck the box next to "Manage repository".
6. Under the "Issues permissions" section, uncheck the box next to "Edit issues".
7. Click "Save changes" to update the write role's permissions.

With these settings, only maintainers will be able to manage the repository, perform actions, and edit issues, while the write role will be restricted to only creating and commenting on issues.

Hope this helps! 🙌🙌

**Reply by BenjaminYde** [reply]
Hi, thanks for the detailed comment, but this is not what I meant.
What you propose is for github repositories, I'm having an access problem with github projects.

As you can see on the picture below the setting is write, but this means that every member can also add views (table,board,...) / change them, ...

![image](https://user-images.githubusercontent.com/44135938/231676914-5c703ab9-8865-4718-887f-5e2408cb9678.png)

What I want is the following:

- Members can be added to the project
- Members can add issues inside the project (and update them)
- Members cannot add / edit views (only maintainers should be able to)
